### ISSUE TITLE
Pact provider reports when there are multiple consumers tested

### ISSUE THREAD

**Issue by anddreiu** [pact-foundation/pact-jvm#1128]
I have multiple consumers (7, at the moment) for the same provider and I am running all the provider tests in a test suite using Junit 4.

The generated Pact provider report (the markdown report) contains currently 130 interactions and it's a little bit hard to search in it if there are failed tests.

It would be a great enhancement if the report could be improved a bit, maybe to have like a summary of all the interactions run based on the consumer names or to group the interactions based on the consumer and then to have all the run interactions.

Thanks!

**Comment by anto-ac** [comment]
@anddreiu Which version of `pact-jvm` are you using? This side of things has been recently improved.

**Comment by anddreiu** [comment]
@anto-ac I am using pact provider junit 4.1.1

**Comment by uglyog** [comment]
A summary would be a really useful feature.

**Comment by anddreiu** [comment]
Hi guys. Are there any updates related to this? It would be really nice and useful to have a summary in the report.
Thanks!

**Comment by uglyog** [comment]
4.1.9 released with the summary added

**Comment by anddreiu** [comment]
Hey @uglyog . Just tried the new version and the summary. It looks like just the first consumer is added in the summary table. Then there are a lot of blank lines and afterwards we have the normal report information added. Please check below the output from an .md report file generated after running the Pact provider tests using Junit runner.

~~~
# providerName
|  Description   |                      Value                      |
|----------------|-------------------------------------------------|
| Date Generated | 2020-10-19T10:06:17.957+03:00[Europe/Bucharest] |
| Pact Version   | 4.1.9                                           |
## Summary
|       Consumer        | Result |
|-----------------------|--------|
| consumerName1         | OK     |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
|                       |        |
## Verifying a pact between _consumerName1_ and _providerName_
~~~

**Comment by uglyog** [comment]
How many consumers do you have?

**Comment by anddreiu** [comment]
I have 8 consumers for the same provider.

**Comment by uglyog** [comment]
@anddreiu version 4.1.10 has been released with the fix

**Comment by anddreiu** [comment]
Thanks @uglyog. I've just upgraded the version in my project and tested this. Now the summary looks good, but the results are not quite alright. I always get OK result even if there are failed interactions for few of the consumers.

**Comment by anddreiu** [comment]
I can see that the summary contains correct result when a provider state is failed: `State change call failed`. But it has to show also failed message when the actual test failed (when the status code is not alright or when matching body or rules are failed)

**Comment by uglyog** [comment]
If the provider state change has failed, it will not attempt to verify the interaction so there will be no message to display for that.

**Comment by anddreiu** [comment]
That case is alright. The problem appears when there isn’t a provider state and the interaction fails, this is not reported ok in the summary.

**Comment by uglyog** [comment]
If the interaction fails, why would you want it to report it as ok in the summary?

**Comment by anddreiu** [comment]
I was misunderstood. I wanted to say that if the interaction fails it is not reported correctly in the summary (it should be failed, but it's reported to be ok). Not sure how to reproduce this, because for some of the consumers it's reported fine, but for others not. 

I am using now 4.1.13 version.

**Comment by uglyog** [comment]
It might be a concurrency issue. I'll try adding a file lock on the report file to see if that fixes the issue.

**Comment by uglyog** [comment]
4.2.8 is released

**Comment by uglyog** [comment]
Released 4.1.24 and 4.2.9

**Comment by YOU54F** [comment]
Closing as complete, if anyone is encountering similar issues, please raise a new one and you can always reference this one.

Thanks
